language/sygus.tex

```diff
@@ -1596,7 +1736,8 @@ \subsection{Satisfying Syntactic Specifications}
 $( \lambda X_1.\, t_1, \ldots, \lambda X_n.\, t_n )$
 satisfies the syntactic restrictions
 for functions-to-synthesize $( f_1, \ldots, f_n )$
-in conjecture $\exists f_1, \ldots, f_n.\, \varphi$
+in conjecture $\exists f_1, \ldots, f_n.\,\forall v_1, \ldots,
```

Comment:
I think it is clearer if you keep the universally quantified variables in that section. 

But regardless, varphi can't be the whole conjecture because in the introduction you say it is a quantifier free formula. You also say that the objective is:

>  to find definitions $e_i$ (in the form of
> expression bodies) for each function $f_i$ such that ...t he constraint
> $\forall v_1, v_2, \ldots, v_m.\ \varphi$ is valid in the background theory
> under the assumption that functions $f_1, \ldots, f_n$ are interpreted as functions returning $e_1, \ldots, e_n$.

so implicitly, \varphi must be  "\varphi_1 \wedge \ldots \wedge \varph_n", although this is never actually stated.

If you want a new symbol, somewhere you need to say that the conjecture is "forall v1 ..vm \varphi_1 \wedge \ldots \varphi_n". Where do you want to say that? Because right now it would need to be in the introduction?

Reply:
Yes, I think you're right that "conjecture" (which I'm calling `\Psi`) needs to be more explicit.

Perhaps the right solution would be to introduce the name `\Psi` for the formula we write at the beginning of Section 3 and then use it here?  I think that would be cleaner so that the definition of overall conjecture is only elaborated in one place.

The point I'm getting at here is that it better to keep the notion of the conjecture more abstract.  By saying here that `\Psi` is some formula (of any form), this means we are not fixing the shape based on the current state of the document.  This means that if we decided to make the possible shape of `\Psi` even more permissive (e.g. nested quantifiers?), this section would not need to be updated.

Reply:
I just noticed you've already defined the synthesis conjecture here, and you've defined it to include the "exists f1,..., fn":
https://github.com/SyGuS-Org/docs/blob/6c71255f345d80080b6091b762a99c61c6b2df22/language/sygus.tex#L536-L538

If that is the synthesis conjecture `Psi`, then replacing `varphi` with `Psi` in the sentence above, gives:
`$\exists f_1, \ldots, f_n.\, \Psi$`
which expands to:
`$\exists f_1, \ldots, f_n.\, \exists f_1, \ldots, f_n. \forall v_1, ...v_m \varphi$`

I suppose technically they are the same, but it's a bit weird. 

So, do I define the synthesis conjecture to be this:
`\Psi = \exists f_1, \ldots, f_n. \forall v_1, ...v_m \varphi$`
or this:
`\Psi =  \forall v_1, ...v_m \varphi$`

Reply:
Yeah, so at the beginning of Section 3, let's say the "conjecture" is
`$\exists f_1, \ldots, f_n.\, \Psi$` where `\Psi` is of the form `\forall v_1, ...v_m (...)`.  Notice that "..."  now includes the constraints + assumptions as you've written.

Then in this section, we can simply consider conjecture `$\exists f_1, \ldots, f_n.\, \Psi$`.

What do you think?